pkg/agents/team5/agent.go

```diff
@@ -0,0 +1,142 @@
+package team5
+
+import (
+	"math"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/food"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/health"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+type Memory struct {
+	foodTaken         food.FoodType //Store last known value of foodTaken by agent
+	agentHP           int           //Store last known value of HP of agent
+	intentionFood     food.FoodType //Store the last known value of the amount of food intended to take
+	favour            int           // e.g. generosity; scale of 0 to 10, with 0 being least favoured and 10 being most favoured
+	daysSinceLastSeen int           // days since last interaction
+}
+
+type CustomAgent5 struct {
+	*infra.Base
+	selfishness       int
+	lastMeal          food.FoodType
+	daysSinceLastMeal int
+	hpAfterEating     int
+	currentAimHP      int
+	attemptFood       food.FoodType
+	satisfaction      int
+	rememberAge       int
+	rememberFloor     int
+	messagingCounter  int
+	// Social network of other agents
+	socialMemory      map[uuid.UUID]Memory
+	surroundingAgents map[int]uuid.UUID
+}
+
+func New(baseAgent *infra.Base) (infra.Agent, error) {
+	return &CustomAgent5{
+		Base:              baseAgent,
+		selfishness:       10, // of 0 to 10, with 10 being completely selfish, 0 being completely selfless
+		lastMeal:          0,  //Stores value of the last amount of food taken
+		daysSinceLastMeal: 0,  //Count of how many days since last eating
+		hpAfterEating:     baseAgent.HealthInfo().MaxHP,
+		currentAimHP:      baseAgent.HealthInfo().MaxHP,
+		attemptFood:       0,
+		satisfaction:      0,                          // Scale of -3 to 3, with 3 being satisfied and unsatisfied
+		rememberAge:       -1,                         // To check if a day has passed by our age increasing
+		socialMemory:      make(map[uuid.UUID]Memory), // Memory of other agents, key is agent id
+		surroundingAgents: make(map[int]uuid.UUID),    // Map agent IDs of surrounding floors relative to current floor
+	}, nil
+}
+
+func (a *CustomAgent5) updateAimHP() {
+	a.currentAimHP = a.HealthInfo().MaxHP - ((10 - a.selfishness) * ((a.HealthInfo().MaxHP - a.HealthInfo().WeakLevel) / 10))
+}
+
+func (a *CustomAgent5) updateSelfishness() {
+	a.selfishness = 10 - a.calculateAverageFavour()
```

Comment:
I see you intended for selfishness to be 0 to 10, would this ensure it's never out of range?

Reply:
Yes, favour is bounded between 0 and 10 inclusive, so calculateAverageFavour can't take any values outside of this range.